updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -36,43 +34,47 @@ def initialize(service:, job:, dependency_snapshot:, error_handler:)
           @job = job
           @dependency_snapshot = dependency_snapshot
           @error_handler = error_handler
-          # This is a placeholder for a real rule object obtained from config in future
-          @dependency_group = Dependabot::DependencyGroup.new(name: GROUP_NAME_PLACEHOLDER, rules: [])
         end
 
+        # rubocop:disable Metrics/AbcSize
         def perform
-          Dependabot.logger.info("[Experimental] Starting grouped update job for #{job.source.repo}")
-          # We should log the rule being executed, let's just hard-code wildcard for now
-          # since the prototype makes best-effort to do everything in one pass.
-          Dependabot.logger.info("Starting update group for '#{GROUP_NAME_PLACEHOLDER}'")
-          dependency_change = compile_all_dependency_changes
-
-          if dependency_change.updated_dependencies.any?
-            Dependabot.logger.info("Creating a pull request for '#{GROUP_NAME_PLACEHOLDER}'")
-            begin
-              service.create_pull_request(dependency_change, dependency_snapshot.base_commit_sha)
-            rescue StandardError => e
-              # FIXME: This is a workround for not having a single Dependency to report against
-              #
-              #        We could use all_updated_deps.first, but that could be misleading. It may
-              #        make more sense to handle the dependency group as a Dependancy-ish object
-              group_dependency = OpenStruct.new(name: "group-all")
-              raise if ErrorHandler::RUN_HALTING_ERRORS.keys.any? { |err| e.is_a?(err) }
-
-              error_handler.handle_dependabot_error(error: e, dependency: group_dependency)
+          # FIXME: This preserves the default behavior of grouping all updates into a single PR
+          # but we should figure out if this is the default behavior we want.
+          register_all_dependencies_group if job.dependency_groups.empty?
```

Comment:
The smoke test seems to be failing on a method missing, we can tolerate nil by flipping the logic to do an `unless &.any?`

```suggestion
          register_all_dependencies_group unless job.dependency_groups&.any?
```

I think this is probably work keeping for now to avoid any co-ordination problems between this PR, the API and smoke tests 👍🏻